### PR TITLE
feat(blogs): more consitency in blog data during save operations

### PR DESCRIPTION
**Pull request by hypeJunction**
Blog attributes and metadata are now consistently updated when blog is saved
irrespective of whether the operation takes part in an action or outside of it.
Auto save action now uses new Ajax API.

BREAKING CHANGE
Response values of the blog/auto_save_revision action have been changed to use
new response API, instead of custom JSON objects.
Status changes to the blog entities that take place outside of the blog/save
action will now be reflected in the river.
ElggBlog class methods have been updated, so overriding classes may need to
reflect these changes

### CODE REVIEW

**Comment by hypeJunction** [comment]
Currently if you do this, you get neither river entries nor notifications:
```php
$blog->status = 'published';
$blog->save();
```

File: mod/blog/actions/blog/auto_save_revision.php [hunk 1]
```diff
@@ -16,78 +16,62 @@
...
+	$blog->status = ElggBlog::UNSAVED_DRAFT;
+	$blog->access_id = ACCESS_PRIVATE;
+	$blog->title = $title;
+	$blog->description = $description;
+	$blog->excerpt = elgg_get_excerpt($excerpt);
 
-		if (!$blog->save()) {
-			$error = elgg_echo('blog:error:cannot_save');
-		}
+	if (!$blog->save()) {
```
Comment (hypeJunction) [inline mod/blog/actions/blog/auto_save_revision.php]: It's bad that we allow blogs to be saved even though the container may change in the save action and prevent the blog from being saved altogether...

**Comment by mrclay** [comment]
FYI I resolved the conflict on this via GitHub just to try it out.

**Comment by mrclay** [comment]
@hypeJunction Is this pretty much ready as is?

**Comment by hypeJunction** [comment]
Should be

**Comment by hypeJunction** [comment]
Revived in #11081

**Pull request closed without merging**